### PR TITLE
bpo-9566: Fix compiler warnings in peephole.c

### PR DESCRIPTION
**Pull request by jkloth**
Fixes compiler warnings in peephole.c when compiled for 64-bit.

https://bugs.python.org/issue9566

### CODE REVIEW

**Comment by serhiy-storchaka** [comment]
This is a duplicate of #10652.

**Comment by vstinner** [comment]
@jkloth: Would you be ok to close this PR in favor in mine, PR #10652?

File: Python/peephole.c [hunk 1]
```diff
@@ -135,6 +135,13 @@ fold_tuple_on_constants(_Py_CODEUNIT *codestr, Py_ssize_t codelen,
     /* Pre-conditions */
     assert(PyList_CheckExact(consts));
 
+#if SIZEOF_SIZE_T > 4
+    if (PyList_GET_SIZE(consts) >= UINT_MAX-1) {
```
Comment (vstinner) [inline Python/peephole.c:139]: My PR uses "if ((size_t)index >= UINT_MAX - 1) {" which shouldn't emit a compiler warning on 32-bit size_t.

File: Python/peephole.c [hunk 2]
```diff
@@ -357,9 +365,18 @@ PyCode_Optimize(PyObject *code, PyObject* consts, PyObject *names,
...
                            argument when they're not taken (so change the
                            first jump to pop its argument when it's taken). */
-                        h = set_arg(codestr, i, (tgt + 1) * sizeof(_Py_CODEUNIT));
-                        j = opcode == JUMP_IF_TRUE_OR_POP ?
-                            POP_JUMP_IF_TRUE : POP_JUMP_IF_FALSE;
+                        oparg = (tgt + 1) * sizeof(_Py_CODEUNIT);
+#if SIZEOF_SIZE_T > 4
+                        /* only replace if it is not too far away */
+                        if (oparg > UINT_MAX) {
+                            h = -1;
```
Comment (vstinner) [inline Python/peephole.c:372]: I cannot overflow in practice. My PR uses an assertion instead, but also makes sure that the code is smaller than INT_MAX at PyCode_Optimize() entry point (sanity check, it should never occur in pratice, see my comment in my PR ;-)).

File: Python/peephole.c [hunk 3]
```diff
@@ -377,23 +394,34 @@ PyCode_Optimize(PyObject *code, PyObject* consts, PyObject *names,
...
+                } else if (UNCONDITIONAL_JUMP(j)) {
+                    oparg = get_arg(codestr, tgt);
+                    if (j == JUMP_FORWARD) {
+                        h = (tgt + 1) * sizeof(_Py_CODEUNIT);
+                        oparg += h;
+                        /* If the new jump target index is larger than the
+                           eval loop's oparg (unsigned int), no rewritting
+                           can be done. */
+#if SIZEOF_SIZE_T > 4
+                        if (oparg > UINT_MAX)
```
Comment (vstinner) [inline Python/peephole.c:411]: It cannot overflow in practice, again, see my PR.

File: Python/peephole.c [hunk 4]
```diff
@@ -429,7 +457,7 @@ PyCode_Optimize(PyObject *code, PyObject* consts, PyObject *names,
     for (i = 0, nops = 0; i < codelen; i++) {
         assert(i - nops <= INT_MAX);
         /* original code offset => new code offset */
-        blocks[i] = i - nops;
+        blocks[i] = (int)(i - nops);
```
Comment (vstinner) [inline Python/peephole.c:460]: blocks uses unsigned int, not signed int.

**Review by vstinner — CHANGES_REQUESTED** [review]

**Comment by vstinner** [comment]
This PR has multiple issues that I already fixed in my PR. So I close this one, sorry @jkloth. I prefer to focus on a single PR :)

**Pull request closed without merging**